app/services/concerns/xccdf/hosts.rb

```diff
@@ -9,20 +9,40 @@ def save_host
       @host.update!(name: inventory_host['fqdn'])
     end
 
-    def save_profile_host
-      test_result_profile.clone_to(account: @account, host: @host)
+    def host_profile
+      @host_profile ||= test_result_profile.clone_to(
+        account: @account, host: @host
+      )
     end
+    alias save_profile_host host_profile
 
     def report_host
-      @metadata&.dig('fqdn') || @test_result_file.host
+      @metadata&.dig('fqdn') || @test_result_file.test_result.host
+    end
+
+    def associate_rules_from_rule_results
+      ::ProfileRule.import!(
+        ::Rule.where(
+          ref_id: selected_op_rule_results.map(&:id),
+          benchmark_id: @benchmark.id
+        ).pluck(:id).map do |rule_id|
+          ::ProfileRule.new(profile_id: @host_profile.id,
+                            rule_id: rule_id)
+        end, ignore: true
+      )
     end
 
     private
 
     def test_result_profile
       ::Profile.canonical
-               .find_by(ref_id: @test_result_file.test_result.profile_id,
-                        benchmark: @benchmark)
+               .where(ref_id: @test_result_file.test_result.profile_id,
+                      benchmark: @benchmark).first ||
+        ::Profile.find_or_initialize_by(
+          ref_id: @test_result_file.test_result.profile_id,
+          name: @test_result_file.test_result.profile_id,
```

Comment:
I think we can get the profile name and description correctly from 
```
<Profile id='foo'>
  <title>$profilename</title>
  <description>$profiledescription</description>
```

Reply:
If there were a profile with a refId that matched this customized profile refID, we could do that. The reason we have to make this PR in the first place is because there is no 'canonical' profile defined in the benchmark with the custom refId. We don't know the parent profile ID of this customized profile, and there's no way to figure it out with just the result output (we'd need the tailoring file as well).

Reply:
We cannot tell who the parent is, that's right - but it looks like we can tell what the name and description are isn't it? Look for `<Profile><title>` in your sample fixture for example

Reply:
There is no `<Profile id="xccdf_org.ssgproject.content_profile_C2S_customized">` in this benchmark with nested title/description. I'm quite sure we can't get that information without the tailoring file.